Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py

```diff
@@ -0,0 +1,233 @@
+        _LOGGER.debug(f'Clipping a value: bias={bias}, scale={scale}, v={v}')
+
+    return value
+
+def GetUvCoord(size, r, g, b):
+    u = g * lutSize + r
+    v = b
+    return (u, v)
+
+def clamp(v):
+    return max(0.0, min(1.0, v))
+
+shaperPresets = {"Log2-48nits":(-6.5, 6.5),
+                 "Log2-1000nits":(-12.0, 10.0),
+                 "Log2-2000nits":(-12.0, 11.0),
+                 "Log2-4000nits":(-12.0, 12.0)}
+
+operations = {"pre-grading":0, "post-grading":1}
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--i', type=str, required=True, help='input file')
+parser.add_argument('--shaper', type=str, required=True, help='shaper preset. Should be one of \'Log2-48nits\', \'Log2-1000nits\', \'Log2-2000nits\', \'Log2-4000nits\'')
+parser.add_argument('--op', type=str, required=True, help='operation. Should be \'pre-grading\' or \'post-grading\'')
+parser.add_argument('--o', type=str, required=True, help='output file')
+parser.add_argument('-l', dest='write3dl', action='store_true', help='output 3dl file')
+parser.add_argument('-a', dest='writeAsset', action='store_true', help='write out azasset file')
+args = parser.parse_args()
+
+# Check for valid shaper type
+invalidShaper = (0,0)
+invalidOp = -1
+
+shaperLimits = shaperPresets.get(args.shaper, invalidShaper)
+if shaperLimits == invalidShaper:
+    _LOGGER.error("invalid shaper")
+    sys.exit(1)
+op = operations.get(args.op, invalidOp)
+if op == invalidOp:
+    _LOGGER.error("invalid operation")
+    sys.exit(1)
+
+# input validation
+from pathlib import Path
+input_file = Path(args.i)
+if input_file.is_file():
+    # file exists
+    pass
+else:
+    FILE_ERROR_MSG = f'File does not exist: {input_file}'
+    _LOGGER.error(FILE_ERROR_MSG)
+    #raise FileNotFoundError(FILE_ERROR_MSG)
+    sys.exit(1)
+
+# Read input image
+#buf = oiio.ImageBuf("linear_lut.exr")
+buf = oiio.ImageBuf(args.i)
+inSpec = buf.spec()
+_LOGGER.info(f"Resolution is x:{inSpec.height}, y:{inSpec.width}")
+
+if inSpec.height < 16:
+    _LOGGER.info(f"invalid input file dimensions: x is {inSpec.height}. Expected LUT with height dimension >= 16 pixels")
+    sys.exit(1)
+    
+def log2(x):
+    return (math.log10(x) /
+            math.log10(2))
+
+def is_power_of_two(n):
+    return (math.ceil(log2(n)) == math.floor(log2(n)))
+    
+if not is_power_of_two(buf.spec().height):
+    _LOGGER.info(f"invalid input file dimensions: {buf.spec().height}. Expected LUT dimensions power of 2: 16, 32, or 64 height")
+    sys.exit(1)
+
+elif inSpec.width != inSpec.height * inSpec.height:
+    _LOGGER.info("invalid input file dimensions. Expect lengthwise LUT with dimension W: s*s  X  H: s, where s is the size of the LUT")
+    sys.exit(1)
+
+lutSize = inSpec.height
+
+middleGrey = 0.18
+lowerStops = shaperLimits[0]
+upperStops = shaperLimits[1]
+
+logMin = math.log(middleGrey * math.pow(2.0, lowerStops), 2.0)
+logMax = math.log(middleGrey * math.pow(2.0, upperStops), 2.0)
```

Comment:
This could be simplified to
```
logMiddleGray = math.log(middleGrey, 2.0)
logMin = logMiddleGray + lowerStops
logMax = logMiddleGray + upperStops
```